packages/react-day-picker/src/components/Table/Table.test.tsx

```diff
@@ -0,0 +1,43 @@
+import * as React from 'react';
+
+import { enGB, enUS } from 'date-fns/locale';
+import { customRender, PageObjects } from 'test';
+
+import { Table } from './Table';
+
+const po = new PageObjects(new Date());
+
+describe('It should render weeks at the boundaries of years correctly', () => {
+  test('If the first days of the month start at week 52 from the previous year, week 52 should be rendered on the 1st row', () => {
+    customRender(<Table displayMonth={new Date(2022, 0)} />, { locale: enGB });
+    const week1 = po.weeks[0].querySelectorAll('td');
+    const day6 = week1[5];
+    const day7 = week1[6];
+    expect(day6).toHaveTextContent('11st January (Saturday)');
+    expect(day7).toHaveTextContent('22nd January (Sunday)');
```

Comment:
I would love suggestions on how to structure these tests better -- right now the text content is from an aria-hidden element (`"2"`) and hidden element that is presumably read by screen readers (`"2nd January (Sunday)"`) -- I don't know to what extend that construct in this v8 beta is accessible.

Reply:
Is there a reason the text of the actual rendered dates (the non-hidden elements) isn't preferred here? Are they not actually visible?

Also, is this react testing library?

Reply:
In addition to these tests, have you thought about including snapshot tests? They can be used to verify that the behavior doesn't change in the future https://jestjs.io/docs/snapshot-testing

Reply:
One way to select the cell in a particular role is something like
```js
const table = screen.getByRole('table')
const firstBodyRow = within(table).getAllByRole('row')[2]
within(firstBodyRow).getByRole('cell', {
  name: /3rd january \(monday\)/i
})
```
Since there are a lot of nesting `within` can be useful:
https://testing-library.com/docs/dom-testing-library/api-within/

Reply:
This might also work if you want to select a row with a matching date
```js
const table = screen.getByRole('table')
within(table).getByRole('row', {
  name: /3rd january \(monday\)/i
})
```
The testing playground is a pretty nice tool to play around with selecting elements and gives recommendations
https://testing-library.com/docs/queries/about/#screenlogtestingplaygroundurl

Reply:
Yes -- but then how to assert that particular element is in the right position in the grid?

Reply:
Good point, can we use something like the the row/col indexes?
```js
const table = screen.getByRole('table')
const firstBodyRow = within(table).getAllByRole('row')[2]
const targetDate = within(firstBodyRow).getAllByRole('cell')[0]
```
There might be a better way than this...